nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/RS2DData.java

```diff
@@ -1102,4 +1164,152 @@ public void setSampleSchedule(SampleSchedule sampleSchedule) {
         this.sampleSchedule = sampleSchedule;
     }
 
+    private static void writeRow(Dataset dataset, Vec vec, int[] pt, BufferedOutputStream fOut) throws IOException {
+        if (dataset.getComplex(0)) {
+            vec.makeComplex();
+        } else {
+            vec.makeReal();
+        }
+        dataset.readVector(vec, pt, 0);
+        if (!dataset.getComplex(0)) {
+            vec.makeComplex();
+        }
+        byte[] array = vec.toFloatBytes();
+        fOut.write(array);
+
+    }
+
+    public static void saveToRS2DFile(Dataset dataset, String filePath) throws IOException {
+        try (BufferedOutputStream fOut = new BufferedOutputStream(new FileOutputStream(filePath))) {
+            int nDim = dataset.getNDim();
+            if (dataset.getNDim() == 1) {
+                Vec vec = dataset.readVector(0, 0);
+                if (!vec.isComplex()) {
+                    vec.hft();
+                }
+                byte[] array = vec.toFloatBytes();
+                fOut.write(array);
+            } else {
+                int[] sizes = new int[dataset.getNDim() - 1];
+                for (int i = 1; i < dataset.getNDim(); i++) {
+                    sizes[i - 1] = dataset.getSizeReal(i);
+                }
+                Vec vec = new Vec(dataset.getSizeReal(0), dataset.getComplex(0));
+
+                int[] pt = new int[dataset.getNDim() - 1];
+                pt[0] = dataset.getSizeReal(0);
+
+                while (true) {
+                    int nRows = sizes[0];
+                    if (dataset.getAxisReversed(1)) {
+                        for (int k = 0; k < nRows; k++) {
+                            pt[0] = k;
+                            writeRow(dataset, vec, pt, fOut);
+                        }
+                    } else {
+                        for (int k = nRows - 1; k >= 0; k--) {
+                            pt[0] = k;
+                            writeRow(dataset, vec, pt, fOut);
+                        }
+                    }
+                    boolean done = true;
+                    for (int j = 2; j < nDim; j++) {
+                        pt[j - 1]++;
+                        if (pt[j - 1] >= sizes[j - 1]) {
+                            if (j == (nDim - 1)) {
+                                break;
+                            }
+                            pt[j - 1] = 0;
+                        } else {
+                            done = false;
+                            break;
+                        }
+                    }
```

Comment:
I'm confused by this loop and stop condition.

Reply:
I'm replacing this code with a MultidimensionalCounter (apache commons math).  That class basically does internally what I wrote here.  I use it other places as its a lot simpler to do. I think I initially didn't here because of the need to worry about reversing order of 2nd dim   But we can't really be sure this is working properly till we have several 3D (and higher) SPINit datasets in the test suite.